demisto_sdk/main.py

```diff
@@ -1,11 +1,241 @@
 import sys
+import click
+from pkg_resources import get_distribution
+
 from demisto_sdk.core import DemistoSDK
+from demisto_sdk.common.tools import str2bool
+from demisto_sdk.dev_tools.linter import Linter
+from demisto_sdk.yaml_tools.unifier import Unifier
+from demisto_sdk.yaml_tools.extractor import Extractor
+from demisto_sdk.common.configuration import Configuration
+from demisto_sdk.validation.secrets import SecretsValidator
+from demisto_sdk.validation.file_validator import FilesValidator
+from demisto_sdk.yaml_tools.content_creator import ContentCreator
+from demisto_sdk.common.constants import SCRIPT_PREFIX, INTEGRATION_PREFIX
+
+
+pass_config = click.make_pass_decorator(DemistoSDK, ensure=True)
+
+
+@click.group(invoke_without_command=True, no_args_is_help=True, context_settings=dict(max_content_width=500))
+@click.help_option(
+    '-h', '--help'
+)
+@click.option(
+    '-d', '--env-dir', help='Specify a working directory.'
+)
+@click.option(
+    '-v', '--version', help='Get the demisto-sdk version.',
+    is_flag=True, default=False
+)
+@pass_config
+def main(config, version, env_dir):
+    config.configuration = Configuration()
+    if version:
+        version = get_distribution('demisto-sdk').version
+        print(version)
+
+    if env_dir:
+        config.configuration.env_dir = env_dir
+
+
+@main.command(name="extract",
+              short_help="Extract code, image and description files from a Demisto integration or script yaml file.")
+@click.help_option(
+    '-h', '--help'
+)
+@click.option(
+    '--infile', '-i',
+    help='The yml file to extract from',
+    required=True
+)
+@click.option(
+    '--outfile', '-o',
+    required=True,
+    help="The output dir to write the extracted code/description/image to."
+)
+@click.option(
+    '--yml-type', '-y',
+    help="Yaml type. If not specified will try to determine type based upon path.",
+    type=click.Choice([SCRIPT_PREFIX, INTEGRATION_PREFIX])
+)
+@click.option(
+    '--demisto-mock', '-d',
+    help="Add an import for demisto mock, true by default",
+    type=click.Choice(["True", "False"]),
+    default=True
+)
+@click.option(
+    '--common-server', '-c',
+    help="Add an import for CommonServerPython."
+         "If not specified will import unless this is CommonServerPython",
+    type=click.Choice(["True", "False"]),
+    default='False'
+)
+@pass_config
+def extract(config, **kwargs):
+    extractor = Extractor(configuration=config.configuration, **kwargs)
+    return extractor.extract_to_package_format()
```

Comment:
it should also be `run`

Reply:
does it really matter, i'm doing two different command with the same class which will make me pass another arg to run, i think that it's better this way